projects/ADuCM3029_demo_cn0414/src/cn0414.c

```diff
@@ -0,0 +1,3611 @@
+	(uint8_t *)"hart_disable",
+	(uint8_t *)"hd",
+	(uint8_t *)"hart_change_channel ",
+	(uint8_t *)"hcc ",
+	(uint8_t *)"hart_transmit ",
+	(uint8_t *)"ht ",
+	(uint8_t *)"hart_get_rec",
+	(uint8_t *)"hg",
+	(uint8_t *)"adc_read_reg ",
+	(uint8_t *)"arr ",
+	(uint8_t *)"adc_write_reg ",
+	(uint8_t *)"awr ",
+	(uint8_t *)"adc_get_samples ",
+	(uint8_t *)"ags ",
+	(uint8_t *)"adc_set_odr ",
+	(uint8_t *)"aso ",
+	(uint8_t *)"adc_set_filt ",
+	(uint8_t *)"asf ",
+	(uint8_t *)"set_update_rate ",
+	(uint8_t *)"sur ",
+	(uint8_t *)"status",
+	(uint8_t *)"stts",
+	(uint8_t *)"adc_set_postfilt ",
+	(uint8_t *)"asp ",
+	(uint8_t *)"adc_en_postfilt",
+	(uint8_t *)"aep",
+	(uint8_t *)"adc_dis_postfilt",
+	(uint8_t *)"adp",
+	(uint8_t *)"adc_set_out_coding ",
+	(uint8_t *)"asoc ",
+	(uint8_t *)"discover_eeprom",
+	(uint8_t *)"de",
+	(uint8_t *)"hart_command_zero ",
+	(uint8_t *)"hcz ",
+	(uint8_t *)"adc_open_wire_enable",
+	(uint8_t *)"aowe",
+	(uint8_t *)"adc_open_wire_disable",
+	(uint8_t *)"aowd",
+	(uint8_t *)"hart_phy_test",
+	(uint8_t *)"hpt",
+	(uint8_t *)"test_routine",
+	(uint8_t *)"tr",
+	(uint8_t *)""
+};
+
+/* Functions for available CLI commands */
+cmd_func v_cmd_fun[] = {
+	cn0414_help,
+	cn0414_channel_display,
+	cn0414_hart_enable,
+	cn0414_hart_disable,
+	cn0414_hart_change_channel,
+	cn0414_hart_transmit,
+	cn0414_hart_get_rec,
+	cn0414_adc_read_reg,
+	cn0414_adc_write_reg,
+	cn0414_adc_get_samples,
+	cn0414_adc_set_odr,
+	cn0414_adc_set_filter,
+	cn0414_channel_set_update_rate,
+	cn0414_status,
+	cn0414_adc_set_postfilt,
+	cn0414_adc_en_postfilt,
+	cn0414_adc_dis_postfilt,
+	cn0414_adc_set_output_coding,
+	cn0414_mem_display_addr,
+	cn0414_hart_send_command_zero,
+	cn0414_adc_open_wire_enable,
+	cn0414_adc_open_wire_disable,
+	cn0414_hart_phy_test,
+	cn0414_system_test_routine,
+	NULL
+};
+
+/* CLI command sizes */
+uint8_t command_size[] = {5, 2, 5, 2, 12, 3, 13, 3, 20, 4, 14, 3, 13, 3, 13, 4,
+			  14, 4, 16, 4, 12, 4, 13, 4, 16, 4, 7, 5, 17, 4, 16, 4, 17, 4, 19,
+			  5, 15, 3, 18, 4, 21, 5, 22, 5, 14, 4, 13, 3, 1
+			 };
+
+/* HART Command zero */
+uint8_t hart_command_zero[] = {0x02, 0x80, 0x00, 0x00, 0x82};
+
+/******************************************************************************/
+/************************ Functions Definitions *******************************/
+/******************************************************************************/
+
+/**
+ * Check the ADC ID register and display error if the correct value is not
+ * found.
+ *
+ * @param [in] dev - The device structure.
+ *
+ * @return 0 in case of success, negative error code otherwise.
+ */
+static int32_t cn0414_setup_adc_verify_id(struct cn0414_dev *dev)
+{
+	ad717x_st_reg *preg;
+	int32_t ret;
+	uint8_t buffer[20];
+
+	preg = AD717X_GetReg(dev->ad4111_device, AD717X_ID_REG);
+
+	if((preg->value & AD717X_ID_REG_MASK) != AD4111_ID_REG_VALUE) {
+		ret = usr_uart_write_string(dev->uart_descriptor,
+					    (uint8_t*)"Error. ADC not found.\n");
+		if(ret != 0)
```

Comment:
for consistency, use `if (ret != SUCCESS) `or something. Check this everywhere in your code